### PR TITLE
 Add aten mkldnn backward ops: max_pool2d, avg_pool2d and adaptive_av…

### PR DESCRIPTION
**Pull request by XiaobingSuper**
### mkldnn backward ops list:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20567) Add aten mkldnn conv2d backward operator :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20570) Add aten mkldnn backward ops: relu, linear and reshape :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20571) Add aten mkldnn backward ops: max_pool2d, avg_pool2d and adaptive_avg_poo2d :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20572) Add aten mkldnn batchnorm backward operator :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20573) Add aten mkldnn zero_ operator💚 
- [ ] \(https://github.com/pytorch/pytorch/pull/20575) Add mkldnn mul operator 💚

### CODE REVIEW

**Comment by Jianhui-Li** [comment]
@gottbrath  @bddppq  @dzhulgakov  @wesolwsk  @VitalyFedyunin @ezyang This batch of PRs enabled the MKL-DNN for the training path and is expected to improve training models like resnext101 by 2x on CPU with AVX512 (also benefit AVX2). With more DL accelerating instructions like bfp16 in CPU roadmap, we think it is important to accelerate the Pytorch training path as we do for the inference path so the community get the benefits.

**Comment by dzhulgakov** [comment]
Hey, @XiaobingSuper - is it possible to rebase these PRs on top of current master and make sure the CI passes?

**Comment by dariogarcia** [comment]
My research group is really interested in this. We are using PyTorch on CPUs because our research has extreme memory requirements, and TensorFlow has a limited tensor size whereas PyTorch does not.

**Comment by gbaydin** [comment]
Hi, our group is interested in this PR to improve CPU training in our large-scale probabilistic programming work.

**Comment by wbhimji** [comment]
I'd like to also support that we need these optimizations for the PyTorch we run in support of large-scale science at the NERSC supercomputing center. Currently we blocking on using more recent PyTorch 1.2 and 1.3 for some projects because of this. Thanks!

**Comment by XiaobingSuper** [comment]
@dariogarcia, @gbaydin, @wbhimji, thank you for your interest in our work. I will rebase the code after get the reponse from Facebook team.

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin

**Comment by DavidRuhe** [comment]
We at SURF (national super-computing facilities in the Netherlands) are also really interested in better MKL support in PyTorch. In our case specifically 3D operations.

**Comment by XiaobingSuper** [comment]
@VitalyFedyunin

**Review by VitalyFedyunin — COMMENTED** [review]
Overall looks good, can you please rebase for final tests. Also what is the plan to start renaming functions to onednn?

**Comment by XiaobingSuper** [comment]
> Overall looks good, can you please rebase for final tests. Also what is the plan to start renaming functions to onednn?

@VitalyFedyunin , the renaming function will be ok at next step which will upgrade DNNL to v1.4, so I will rebase those PRs which stiil using the old name, thanks！
